dlt/common/configuration/specs/base_configuration.py

```diff
@@ -252,6 +252,13 @@ def wrap(cls: Type[TAnyClass]) -> Type[TAnyClass]:
                         )
                     setattr(cls, att_name, None)
 
+                from types import MappingProxyType
+
+                # make dict immutable: TODO: fix propery
+                # somehow default_factory does not work here..
+                if isinstance(att_value, dict):
```

Comment:
we are creating dataclasses with a mutable dict value in some cases (when the default value is an assigned dict, this happens in one of the verified sources and will fail the tests where all verified sources are created). I tried to use the default_factory workaround from below, but somehow it did not work. Changing the value into something immutable fixes this, but I'm not sure this is right. Will also happen for lists.